### PR TITLE
Add 2022 code style

### PR DESCRIPTION
**Pull request by weeman1337**
## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/vector-im/element-web/blob/develop/CONTRIBUTING.md))

## PR list

* https://github.com/vector-im/element-web/pull/23153
* https://github.com/matrix-org/matrix-js-sdk/pull/2621
* https://github.com/matrix-org/matrix-react-sdk/pull/9220
* https://github.com/matrix-org/eslint-plugin-matrix-org/pull/30

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,4 +1,282 @@
-Contributing code to Element
-============================
+Contributing code to Element Web
```
Comment (weeman1337) [inline CONTRIBUTING.md:1]: Basically a copy of https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md with `s/matrix-js-sdk/Element Web` .

File: code_style.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
+# Matrix Code Style Guide
```
Comment (weeman1337) [inline code_style.md:1]: Comes from here https://github.com/vector-im/element-meta/pull/216/files

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,4 +1,282 @@
...
+================================
 
-Element follows the same pattern as the [matrix-js-sdk](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md).
+Everyone is welcome to contribute code to Element Web, provided that they are
```
Comment (weeman1337) [inline CONTRIBUTING.md:4]: Basically a copy of https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md with `s/matrix-js-sdk/Element Web` .

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,4 +1,282 @@
...
+Tests
+-----
+Your PR should include tests.
+
+For new user facing features in `matrix-react-sdk` or `element-web`, you
+must include:
+
+1. Comprehensive unit tests written in Jest. These are located in `/test`.
+2. "happy path" end-to-end tests.
+   These are located in `/test/end-to-end-tests` in `matrix-react-sdk`, and
```
Comment (duxovni) [inline CONTRIBUTING.md:116]: This should be updated for Cypress

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,4 +1,282 @@
...
+
+We use continuous integration, and all pull requests get automatically tested:
+if your change breaks the build, then the PR will show that there are failed
+checks, so please check back after a few minutes.
+
+Tests
+-----
+Your PR should include tests.
+
+For new user facing features in `matrix-react-sdk` or `element-web`, you
```
Comment (duxovni) [inline CONTRIBUTING.md:111]: This should also reference `matrix-js-sdk` now that it's not part of that repo

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,4 +1,282 @@
...
+functionality. End-to-end tests should be added prior to the feature
+leaving labs, but don't have to be present from the start (although it might
+be beneficial to have some running early, so you can test things faster).
+
+For bugs in those repos, your change must include at least one unit test or
+end-to-end test; which is best depends on what sort of test most concisely
+exercises the area.
+
+Changes to must be accompanied by unit tests written in Jest.
+These are located in `/spec/`.
```
Comment (duxovni) [inline CONTRIBUTING.md:132]: This is js-sdk specific

File: code_style.md (hunk 2 above)
Comment (duxovni) [inline code_style.md:1]: Should this be titled "Element Web Style Guide" given that it's specifically for the element-web codebase?

**Review by duxovni — CHANGES_REQUESTED** [review]

File: code_style.md (hunk 2 above)
Comment (weeman1337) [reply code_style.md:1]: Copied the description from the code style PR

**Review by duxovni — APPROVED** [review]

**Pull request merged**